_episodes/05-preprocessor.md

```diff
@@ -0,0 +1,287 @@
+- "Can I use different datasets for different variables?"
+objectives:
+- "Create a recipe with multiple preprocessors"
+- "Use different preprocessors for different variables"
+- "Run a recipe with variables from different datasets"
+keypoints:
+- "A recipe can run different preprocessors at the same time."
+- "The setting additional_datasets can be used to add a different dataset."
+- "Variable groups are useful for defining different settings for different variables."
+---
+
+## Preprocessors: What are they and how do they work?
+
+Preprocessing is the process of performing a set of modular operations on the data before applying diagnostics or metrics. In the figure below you see the preprocessor functions in the light blue box on the right.
+
+![figure showing ESMValTool architecture]({{ page.root }}/fig/esmvaltool_architecture.png)
+
+Underneath the hood, each preprocessor is a modular python function that receives an iris cube and sometimes some arguments. The preprocessor applies some mathematical or computational transformation to the input iris cube, then returns the processed iris cube.
+
+## Inspect the example preprocessor
+
+Each preprocessor section includes a preprocessor name, a list of preprocessor steps to be executed and any arguments needed by the preprocessor steps.
+
+>~~~YAML
+> preprocessors:
+>   prep_timeseries:
+>     annual_statistics:
+>       operator: mean
+>~~~
+{: .source}
+
+For instance, the 'annual_statistics' with the  'operation: mean' argument preprocessor receives an iris cube, takes the annual average for each year of data in the cube, and returns the processed cube. 
+
+You could use several preprocessor steps listed in the [documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/preprocessor.html). The standardised interface between the preprocessors allows them to be used modularly - like lego blocks. Almost any conceivable preprocessing order of operations can be performed using ESMValTool preprocessors. 
+
+> ## The 'custom order' command.
+>
+>If you do not want your preprocessors to be applied in the [default order](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/api/esmvalcore.preprocessor.html), then add the following line to your preprocessor chain:
+>
+>~~~
+>    custom_order: true
+>~~~
+>
+>The default preprocessor order is listed in the [ESMValCore preprocessor API page](>https://docs.esmvaltool.org/projects/ESMValCore/en/latest/api/esmvalcore.preprocessor.html).
+>
+> Also note that not preprocessor operations aren't always commutative - meaning that the order of operations matters. For instance, if you run the preprocessor 'extract_volume' to extract the top 100m of the ocean surface, then 'volume_statistics' to calculate the volume-weighted mean of the data, then your result will differ depending on the order of these two preprocessors. In fact, the 'extract_volume' preprocessor will fail if you try to run it on a 2D dataset.
```

Comment:
There is something off in the first half of the sentence. Either rephrase to "Also note that not all preprocessor operations are commutative..." or "Also note that preprocessor operations arent't always commutative..."

Reply:
Fixed that and the next sentence as well.